mailpoet/lib/WooCommerce/Helper.php

```diff
@@ -62,7 +62,7 @@ public function wcGetProduct($theProduct = false) {
     return wc_get_product($theProduct);
   }
 
-  public function wcGetPageId(string $page): ?int {
+  public function wcGetPageId(string $page) {
     if (function_exists('wc_get_page_id')) {
       return wc_get_page_id($page);
     }
```

Comment:
> To prevent this from being a problem, we could either remove the ?int return type hint or always coerce the value returned from wc_get_page_id to be an integer.

May I ask why you decided not to go with the second option (always coerce the value returned from wc_get_page_id to be an integer)?

Reply:
Of course! Honestly I was torn between the two options. Removing the type hint felt a tiny bit more appropriate to me because if we do find ourselves in an edge case where `wc_get_page_id` is defined by something other than WooCommerce and returns something other than an integer, coercing the value would mean our wrapper would return a different type than the wrapped function, which just feels inconsistent. Not having the return type hint makes it clearer to me that we can't actually be 100% sure what `wc_get_page_id` will return.

Would your preference be for keeping the type hint and coercing the value in the wrapper?

Reply:
> we do find ourselves in an edge case where wc_get_page_id is defined by something other than WooCommerce and returns something other than an integer

We should not return either string or integer if WooCommerce is not installed.  
The function is meant to get a WooCommerce page ID by name. If WooCommerce is not installed and activated, we should return null.

That would prevent some weird bugs down the road.

If wc_get_page_id was defined by something other than WooCommerce itself, there is a high chance the behavior would be different.



> Would your preference be for keeping the type hint and coercing the value in the wrapper?

Yes. 

We won’t have to coerce the function return value everywhere else it's been used. e.g. like you did [here](https://github.com/mailpoet/mailpoet/blob/20f79c45cf4f8ccf65b0db94ddfd15f2f51b9a01/mailpoet/lib/Form/DisplayFormInWPContent.php#L126)

Reply:
@triple0t I ended up going a slightly different route. I replaced the check to see if the function `wc_get_page_id` is defined to instead check if WC is installed and active, which should ensure that the version of `wc_get_page_id` that exists actually comes from WC. I left in the return type hint, and I also included a type cast just in case something changes in WC's implementation in the future. Does this seem reasonable to you?

One thing I struggled with while thinking about this is consistency. Many of our other wrapped WC functions (`getWoocommerceCurrency`, `get_woocommerce_currency_symbol`, `wcLightOrDark` etc) don't include return type hints, and they don't even include checks for the existence of the function or if WC is active. Instead it looks like we either use these functions in contexts where we know WC must be active (like WC hooks), or we include checks to `isWooCommerceActive` in the application code before we call the helper. 

Part of me thinks the existing convention makes a little more sense. Why call these methods in the first place unless we know WC is installed/active? If we return `null` if WC isn't active that's just one more thing we might have to check for depending on what the original signature looked like.

In any case, I'm probably overthinking this and spending too much time writing this comment, but I do think it's worth thinking about what convention makes the most sense for these wrapped functions in general.

Reply:
> @triple0t I ended up going a slightly different route. I replaced the check to see if the function wc_get_page_id is defined to instead check if WC is installed and active, which should ensure that the version of wc_get_page_id that exists actually comes from WC. I left in the return type hint, and I also included a type cast just in case something changes in WC's implementation in the future. Does this seem reasonable to you?

Yup. It does.

> One thing I struggled with while thinking about this is consistency. Many of our other wrapped WC functions (getWoocommerceCurrency, get_woocommerce_currency_symbol, wcLightOrDark etc) don't include return type hints, and they don't even include checks for the existence of the function or if WC is active. Instead it looks like we either use these functions in contexts where we know WC must be active (like WC hooks), or we include checks to isWooCommerceActive in the application code before we call the helper.
>
> Part of me thinks the existing convention makes a little more sense. Why call these methods in the first place unless we know WC is installed/active? If we return null if WC isn't active that's just one more thing we might have to check for depending on what the original signature looked like.
>
> In any case, I'm probably overthinking this and spending too much time writing this comment, but I do think it's worth thinking about what convention makes the most sense for these wrapped functions in general.

I definitely agree with you.